modules/gdscript/gdscript_editor.cpp

```diff
@@ -2004,8 +2004,8 @@ static bool _guess_identifier_type(GDScriptParser::CompletionContext &p_context,
 		return false;
 	}
 
-	// Check autoloads.
-	if (ProjectSettings::get_singleton()->has_autoload(p_identifier)) {
+	// Check autoloads. Identifier may not be in the named globals map if "Global Variable" option is disabled.
+	if (ProjectSettings::get_singleton()->has_autoload(p_identifier) && GDScriptLanguage::get_singleton()->get_named_globals_map().has(p_identifier)) {
```

Comment:
If you're checking the `get_named_globals_map()` you don't need to check the autoloads in ProjectSettings, it's redundant (named globals are only used for autoloaded singletons anyway).

Reply:
Will we ever implement globals that aren't Autoloads?

Reply:
It probably doesn't matter in that case, the "autoload" part of the comment could be replaced by "global variables".

Reply:
Looks great now :slightly_smiling_face: